tests/integration/test_permissions.py

```diff
@@ -28,324 +28,125 @@
-    inspire_recid_minter(record.id, record)
-    db.session.commit()
-    es.indices.refresh('records-hep')
-
-    return record
-
-
-@pytest.fixture(scope='function')
-def sample_record(app):
-    record = {
-        '$schema': 'http://localhost:5000/schemas/records/hep.json',
-        '_collections': [
-            'Literature',
-        ],
-        'control_number': 111,
-        'document_type': [
-            'article',
-        ],
-        'titles': [
-            {'title': 'sample'}
-        ],
-    }
-    record = _create_and_index_record(record)
-    record_id = record.id
-
-    yield record
-
-    pid = PersistentIdentifier.get('lit', '111')
-    db.session.delete(pid)
-    record = InspireRecord.get_record(record_id)
-    record._delete(force=True)
-    current_app.extensions[
-        'invenio-db'].versioning_manager.transaction_cls.query.delete()
-    db.session.commit()
-
-
-@pytest.fixture(scope='function')
-def restricted_record(app):
-    restricted_collection = Collection(
-        name='Restricted Collection',
-        dbquery='_collections:"Restricted Collection"',
-    )
-    another_restricted_collection = Collection(
-        name='Another Restricted Collection',
-        dbquery='_collections:"Another Restricted Collection"',
-    )
-
-    db.session.add_all([restricted_collection, another_restricted_collection])
-    db.session.commit()
-
-    record = {
-        '$schema': 'http://localhost:5000/schemas/records/hep.json',
-        '_collections': [
-            'Literature',
-            'Restricted Collection',
-            'Another Restricted Collection',
-        ],
-        'control_number': 222,
-        'document_type': [
-            'article',
-        ],
-        'titles': [
-            {'title': 'restricted'}
-        ],
-    }
-
-    record = _create_and_index_record(record)
-    yield record
-
-    collection = Collection.query.filter_by(
-        name='Restricted Collection'
-    ).one()
-    another_collection = Collection.query.filter_by(
-        name='Another Restricted Collection'
-    ).one()
-    pid = PersistentIdentifier.get('lit', '222')
-    db.session.delete(collection)
-    db.session.delete(another_collection)
-    db.session.delete(pid)
-    record._delete(force=True)
-    current_app.extensions[
-        'invenio-db'].versioning_manager.transaction_cls.query.delete()
-    db.session.commit()
-    current_cache.delete('restricted_collections')
-
-
-@pytest.fixture(scope='function')
-def users(app):
-    # Create test users
-    hashed_password = hash_password('123456')
-    user = User(
-        email='user@inspirehep.net',
-        password=hashed_password,
-        active=True
-    )
-    user_partially_allowed = User(
-        email='partially_allowed@inspirehep.net',
-        password=hashed_password,
-        active=True,
-    )
-    user_allowed = User(
-        email='allowed@inspirehep.net',
-        password=hashed_password,
-        active=True,
-    )
```

Comment:
There's other places in which this refactoring has to happen. Grep for `users` in `tests` and you should find several other examples.